rosserial_test/src/publish_subscribe.cpp

```diff
@@ -47,6 +47,28 @@ static void rosserial_string_cb(const rosserial::std_msgs::String& msg)
   last_string = std::string(msg.data);
 }
 
+class NonStaticInstance
+{
+public:
+  NonStaticInstance():
+    rosserial_string_cb_count_(0)
+  {}
+  ~NonStaticInstance() {}
+
+  void rosserial_string_cb(const rosserial::std_msgs::String& msg)
+  {
+    rosserial_string_cb_count_++;
+    last_string_ = std::string(msg.data);
+  }
+
+  int getCount() { return rosserial_string_cb_count_; }
+  std::string getLastString() { return last_string_; }
+
+private:
+  int rosserial_string_cb_count_;
+  std::string last_string_;
+};
```

Comment:
May not be necessary, if we can just reuse the one from here: https://github.com/ros-drivers/rosserial/blob/jade-devel/rosserial_test/include/rosserial_test/helpers.h#L3